openapi_python_client/parser/properties/__init__.py

```diff
@@ -250,21 +250,25 @@ def build_model_property(
     required_properties: List[Property] = []
     optional_properties: List[Property] = []
     relative_imports: Set[str] = set()
-    references: List[oai.Reference] = []
 
     class_name = data.title or name
     if parent_name:
         class_name = f"{utils.pascal_case(parent_name)}{utils.pascal_case(class_name)}"
     ref = Reference.from_ref(class_name)
 
     all_props = data.properties or {}
-    if not isinstance(data, oai.Reference) and data.allOf:
-        for sub_prop in data.allOf:
-            if isinstance(sub_prop, oai.Reference):
-                references += [sub_prop]
-            else:
-                all_props.update(sub_prop.properties or {})
-                required_set.update(sub_prop.required or [])
+    for sub_prop in data.allOf or []:
+        if isinstance(sub_prop, oai.Reference):
+            source_name = Reference.from_ref(sub_prop.ref).class_name
+            sub_model = schemas.models.get(source_name)
+            if sub_model is None:
+                return PropertyError(f"Reference {sub_prop.ref} not found"), schemas
+            required_properties.extend(sub_model.required_properties)
+            optional_properties.extend(sub_model.optional_properties)
```

Comment:
I think you need to account for a situation where one of the children of `allOf` specify a property as required and one does not (in which case should make it should be required)

Reply:
Yeah you're right. I think attrs will implement `__eq__` and maybe even `__hash__` so we can remove any optional properties from the required properties list. I think we can also get duplicate properties with the current implementation.

What do you think we should do if two properties of the same name are defined but not the same? Return an error, right?

Reply:
Depends on why they're not the same, I think. If there are things in direct conflict (like different `type`s) then error. But if it's one has it required, the other optional, then just make it required.

Different sub-properties in 2 different model props could also probably error